### PR TITLE
applications: nrf_desktop: Add SMP SUIT DFU support for nrf54h20

### PR DESCRIPTION
**Pull request by zycz**
Add SMP SUIT DFU support for nrf_desktop application.

JIRA: NCSDK-26998

### CODE REVIEW

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 1]
```diff
@@ -127,6 +134,7 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			cast_module_state_event(aeh);
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+#if defined CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
```
Comment (kapi-no) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:137]: L137-L168, replace with a single line:

```dfu_backend_init();```

And then define two variants of this function for each backend:

```
#if CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
static void dfu_backend_init(void)
{
	if (!IS_ENABLE(CONFIG_DESKTOP_DFU_MCUMGR_MCUBOOT_DIRECT_XIP) {
		int err = boot_write_img_confirmed();

		if (err) {
			LOG_ERR("Cannot confirm a running image: %d", err);
		}
	}

	LOG_INF("MCUboot image version: %s", CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION);
}
#elif CONFIG_DESKTOP_DFU_BACKEND_SUIT
static void dfu_backend_init(void)
{
	/* Manifest class ID
	 * nRF Desktop uses default SUIT manifest file (default namespace and name).
	 */
	const suit_manifest_class_id_tmanifest_class_id = {{
		/* RFC4122 uuid5(uuid5(uuid.NAMESPACE_DNS, 'nordicsemi.com'),
		 *               'nRF54H20_sample_root')
		 */
		0x3f, 0x6a, 0x3a, 0x4d, 0xcd, 0xfa, 0x58, 0xc5,
		0xac, 0xce, 0xf9, 0xf5, 0x84, 0xc4, 0x11, 0x24
	}};
	unsigned int seq_num = 0;

	int err = suit_get_installed_manifest_info((
				suit_manifest_class_id_t *)&manifest_class_id,
				&seq_num, NULL, NULL, NULL, NULL);
	if (!err) {
		LOG_INF("SUIT image version: %d", seq_num);
	} else {
		LOG_ERR("suit_read_manifest_data failed (err: %d)", err);
	}
}
#else
#error "The DFU backend choice is not supported"
#endif /* CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT */
```

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 2]
```diff
@@ -4,8 +4,15 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#include <zephyr/kernel.h>
+
+#if defined CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
```
Comment (kapi-no) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:9]: I think we don't need this #if defined statement here as these MCUboot headers shouldn't have any impact on the SUIT backend

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 3]
```diff
@@ -4,8 +4,15 @@
...
 #include <zephyr/dfu/mcuboot.h>
 #include <zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt.h>
+#elif defined CONFIG_DESKTOP_DFU_BACKEND_SUIT
```
Comment (kapi-no) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:12]: Let's get rid of this #elif as well

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 4]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
...
+CONFIG_ZCBOR=y
+CONFIG_ZCBOR_CANONICAL=y
+CONFIG_SSF_SUIT_SERVICE_ENABLED=y
+CONFIG_DESKTOP_DFU_MCUMGR_ENABLE=y
+
+CONFIG_MGMT_SUITFU=y
+CONFIG_MCUMGR_SMP_LEGACY_RC_BEHAVIOUR=y
+
+# Enable the Bluetooth MCUmgr transport (unauthenticated).
+CONFIG_MCUMGR=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:104]: This option is unnecessary. The DESKTOP_DFU_MCUMGR_ENABLE configuration already selects it

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 5]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
...
+
+CONFIG_SUIT=y
+CONFIG_SUIT_DFU_CANDIDATE_PROCESSING_MINIMAL=y
+CONFIG_ZCBOR=y
+CONFIG_ZCBOR_CANONICAL=y
+CONFIG_SSF_SUIT_SERVICE_ENABLED=y
+CONFIG_DESKTOP_DFU_MCUMGR_ENABLE=y
+
+CONFIG_MGMT_SUITFU=y
+CONFIG_MCUMGR_SMP_LEGACY_RC_BEHAVIOUR=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:101]: These options are also selected by the `DESKTOP_DFU_BACKEND_SUIT` option

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 6]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
 CONFIG_SERIAL=n
 CONFIG_CONSOLE=n
 CONFIG_UART_CONSOLE=n
+
+################################################################################
+# DFU configuration
+
+CONFIG_SUIT=y
+CONFIG_SUIT_DFU_CANDIDATE_PROCESSING_MINIMAL=y
+CONFIG_ZCBOR=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:95]: `ZCBOR` is already selected by the `DESKTOP_DFU_MCUMGR_ENABLE` option

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 7]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
...
+# Enable the MCUmgr Packet Reassembly feature over Bluetooth and its configuration dependencies.
+# MCUmgr buffer size is optimized to fit one SMP packet divided into five Bluetooth Write Commands,
+# transmitted with the maximum possible MTU value: 498 bytes.
+CONFIG_MCUMGR_TRANSPORT_BT_REASSEMBLY=y
+CONFIG_MCUMGR_TRANSPORT_NETBUF_SIZE=2475
+CONFIG_MCUMGR_TRANSPORT_WORKQUEUE_STACK_SIZE=4096
+
+# Allow a DFU host to retrieve information about the number and size of MCUmgr buffers.
+# The nRF Connect Device Manager uses this information to speed up the DFU.
+CONFIG_MCUMGR_GRP_OS=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:122]: This option is also selected by the `DESKTOP_DFU_MCUMGR_ENABLE` option

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 8]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
...
+CONFIG_MCUMGR_GRP_OS_MCUMGR_PARAMS=y
+
+# Extended SUIT commands over USER group
+CONFIG_MGMT_SUITFU_GRP_SUIT=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:126]: Is this option required for SUIT DFU? Perhaps it could also be selected by the `DESKTOP_DFU_BACKEND_SUIT` option?

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf [hunk 9]
```diff
@@ -86,3 +86,41 @@ CONFIG_BT_CONN_TX_MAX=4
...
 CONFIG_UART_CONSOLE=n
+
+################################################################################
+# DFU configuration
+
+CONFIG_SUIT=y
+CONFIG_SUIT_DFU_CANDIDATE_PROCESSING_MINIMAL=y
+CONFIG_ZCBOR=y
+CONFIG_ZCBOR_CANONICAL=y
+CONFIG_SSF_SUIT_SERVICE_ENABLED=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj_release.conf:97]: These options could potentially also be selected by the `DESKTOP_DFU_BACKEND_SUIT` option

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:1]: Same improvements could be applied for this file as for prj_release.conf

**Review by kapi-no — COMMENTED** [review]
Looking good, suggested a few improvements

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 2 above)
Comment (zycz) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:9]: Then I get error: fatal error: bootutil/image.h: No such file or directory
I should then add some more konfigs to include extra files

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 3 above)
Comment (zycz) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:12]: Can do it, but prefer not to include files which are 100% not used. IMO leaving it as it is much better option.

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 2 above)
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:9]: I see it's an issue with the following header:

https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt.h

that includes the MCUboot repository headers when the relevant Kconfig is enabled

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 3 above)
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:12]: Well, this header is used in the SUIT configuration of this module. It's like `IS_ENABLED` macro, the build system will figure out that this header is not needed in the MCUboot configuration as there is no reference to its APIs

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr [hunk 10]
```diff
@@ -43,6 +39,33 @@ config DESKTOP_DFU_MCUMGR_MCUBOOT_DIRECT_XIP
...
+	help
+	  This option enables the MCUboot bootloader as the backend for the MCUmgr DFU module.
+
+config DESKTOP_DFU_BACKEND_SUIT
+	bool "Use SUIT as DFU backend"
+	select MCUMGR_SMP_LEGACY_RC_BEHAVIOUR
+	select MGMT_SUITFU
+	select ZCBOR_CANONICAL
+	select SSF_SUIT_SERVICE_ENABLED
+	select MGMT_SUITFU_GRP_SUIT
```
Comment (kapi-no) [inline applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:62]: To keep the related options next to each other and keep the alphabetical order

```suggestion
	select MGMT_SUITFU
	select MGMT_SUITFU_GRP_SUIT
	select SSF_SUIT_SERVICE_ENABLED
	select ZCBOR_CANONICAL	
```

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 3 above)
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:12]: I don't insist but let's see what others think

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 2 above)
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:9]: To sum up just the `<zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt.h>` requires the ifdef statement

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 3 above)
Comment (mkapala-nordic) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:12]: If we still need to ifdef the mcuboot part then imo we can leave it as is.

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 11]
```diff
@@ -110,6 +117,47 @@ static struct mgmt_callback cmd_recv_cb = {
...
+	/* Manifest class ID
+	 * nRF Desktop uses default SUIT manifest file (default namespace and name).
+	 */
+	const suit_manifest_class_id_t manifest_class_id = {{
+		/* RFC4122 uuid5(uuid5(uuid.NAMESPACE_DNS, 'nordicsemi.com'),
+		 *              'nRF54H20_sample_root')
+		 */
+		0x3f, 0x6a, 0x3a, 0x4d, 0xcd, 0xfa, 0x58, 0xc5,
+		0xac, 0xce, 0xf9, 0xf5, 0x84, 0xc4, 0x11, 0x24
+	}};
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:145]: I wonder if we could not simplify it by doing the https://nordicsemi.atlassian.net/browse/NCSDK-25024 already here.
It seems it would be just one API call to add here and we already had to add this logic here.
However, I'll won't insist it now as it is in separate ticket.

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/app.overlay [hunk 12]
```diff
@@ -52,3 +52,18 @@
 		};
 	};
 };
+
+&cpuapp_rw_partitions {
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/app.overlay:56]: Shouldn't we configure it somewhat similar to the SMP sample?

https://github.com/nrfconnect/sdk-nrf/blob/main/samples/suit/smp_transfer/sysbuild/nrf54h20dk_nrf54h20_memory_map.dtsi

?

BTW, where is located the storage partition?

Comment (kapi-no) [reply applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/app.overlay:56]: discussed on Teams

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 11 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:145]: Simplifying implementation in scope of this PR may actually be a good idea (change should be really small and easy to apply)

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 13]
```diff
@@ -110,6 +117,47 @@ static struct mgmt_callback cmd_recv_cb = {
...
+	unsigned int seq_num = 0;
+
+	int err = suit_get_installed_manifest_info((
+				suit_manifest_class_id_t *)&manifest_class_id,
+				&seq_num, NULL, NULL, NULL, NULL);
+	if (!err) {
+		LOG_INF("SUIT image version: %d", seq_num);
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:152]: Can we call it SUIT sequence number? To be consistent with how SUIT names it.

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf [hunk 14]
```diff
@@ -106,3 +106,30 @@ CONFIG_LOG_MODE_DEFERRED=y
 CONFIG_LOG_PRINTK=y
 CONFIG_LOG_BUFFER_SIZE=4096
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
+
+################################################################################
+# DFU configuration
+
+CONFIG_SUIT=y
+CONFIG_SUIT_DFU_CANDIDATE_PROCESSING_MINIMAL=y
+CONFIG_DESKTOP_DFU_MCUMGR_ENABLE=y
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:115]: Please move application modules and DFU transport configuration out of `DFU configuration` section. See `prj_fast_pair.conf` of `nrf52840dk` for reference.

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf [hunk 15]
```diff
@@ -106,3 +106,30 @@ CONFIG_LOG_MODE_DEFERRED=y
...
+
+# Enable the Bluetooth MCUmgr transport (unauthenticated).
+CONFIG_MCUMGR_TRANSPORT_BT=y
+CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN=n
+
+# Allow for large Bluetooth data packets.
+CONFIG_BT_L2CAP_TX_MTU=498
+CONFIG_BT_BUF_ACL_RX_SIZE=502
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:123]: I think we need to align Bluetooth buffers configuration on radio core too.

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr [hunk 16]
```diff
@@ -9,20 +9,16 @@ menuconfig DESKTOP_DFU_MCUMGR_ENABLE
 	select EXPERIMENTAL
 	select CAF_BLE_SMP_TRANSFER_EVENTS if MCUMGR_TRANSPORT_BT
 	select MCUMGR
-	select MCUMGR_GRP_IMG
 	select MCUMGR_GRP_OS
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:12]: Do we need `CONFIG_MCUMGR_GRP_OS` for SUIT?

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr [hunk 17]
```diff
@@ -9,20 +9,16 @@ menuconfig DESKTOP_DFU_MCUMGR_ENABLE
...
 	select MCUMGR_MGMT_NOTIFICATION_HOOKS
 	select MCUMGR_GRP_IMG_MUTEX if DESKTOP_DFU_LOCK
 	select MCUMGR_SMP_COMMAND_STATUS_HOOKS
 	select NET_BUF
 	select ZCBOR
 	select CRC
-	select IMG_MANAGER
-	select MCUBOOT_BOOTUTIL_LIB
 	select STREAM_FLASH
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:19]: Is stream flash used by SUIT?  (please also double check the remaining Kconfig options selected as common)

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:9]: Ok, then it seems that we need to use ifdefs (at least for now)

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Comment by kapi-no** [comment]
@zycz, could you resolve Compliance issue?

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr (hunk 16 above)
Comment (zycz) [reply applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:12]: Yes because:

# Allow a DFU host to retrieve information about the number and size of MCUmgr buffers.
# The nRF Connect Device Manager uses this information to speed up the DFU.
CONFIG_MCUMGR_GRP_OS_MCUMGR_PARAMS=y

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:12]: If it's only speed improvement (not an actual requirement) it should be enabled in `*.conf` file I think (or eventually it can be implied). Keep in mind that OS mgmt is required for SMP DFU on nRF52, because it enables system reboot command that is used right after finalized DFU image transfer.

File: applications/nrf_desktop/src/util/desktop_suit.c [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include "desktop_suit.h"
+#include <sdfw/sdfw_services/suit_service.h>
+
+int suit_get_installed_manifest_seq_num(unsigned int *seq_num)
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/src/util/desktop_suit.c:11]: I think that prefix should be `desktop_suit_*` so we will not mistake it with the APIs from the SUIT service.

File: applications/nrf_desktop/src/util/desktop_suit.c [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+	/* Manifest class ID
+	 * nRF Desktop uses default SUIT manifest file (default namespace and name).
+	 */
+	const suit_manifest_class_id_t manifest_class_id = {{
+		/* RFC4122 uuid5(uuid5(uuid.NAMESPACE_DNS, 'nordicsemi.com'),
+		 *              'nRF54H20_sample_root')
+		 */
+		0x3f, 0x6a, 0x3a, 0x4d, 0xcd, 0xfa, 0x58, 0xc5,
+		0xac, 0xce, 0xf9, 0xf5, 0x84, 0xc4, 0x11, 0x24
+	}};
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/src/util/desktop_suit.c:22]: What about https://github.com/nrfconnect/sdk-nrf/pull/15553#discussion_r1618621960 and usage of `suit_get_supported_manifest_info`?

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 20]
```diff
@@ -110,6 +122,36 @@ static struct mgmt_callback cmd_recv_cb = {
...
+static void dfu_backend_init(void)
+{
+	unsigned int seq_num = 0;
+
+	int err = suit_get_installed_manifest_seq_num(&seq_num);
+
+	if (!err) {
+		LOG_INF("SUIT sequence number: %d", seq_num);
+	} else {
+		LOG_ERR("suit_read_manifest_data failed (err: %d)", err);
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:148]: Mismatched function name

File: applications/nrf_desktop/src/util/CMakeLists.txt [hunk 21]
```diff
@@ -20,6 +20,9 @@ endif()
 target_sources_ifdef(CONFIG_DESKTOP_ADV_PROV_UUID16_ALL
 		     app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/bt_le_adv_prov_uuid16.c)
 
+         target_sources_ifdef(CONFIG_DESKTOP_DFU_BACKEND_SUIT
+		     app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/desktop_suit.c)
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/src/util/CMakeLists.txt:24]: unnecessary whitespace

**Review by mkapala-nordic — CHANGES_REQUESTED** [review]

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 22]
```diff
@@ -37,7 +45,11 @@ static void dfu_lock_owner_changed(const struct dfu_lock_owner *new_owner)
 	/* The function declaration is not included in MCUmgr's header file if the mutex locking
 	 * of the image management state object is disabled.
 	 */
+#if defined CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
 	img_mgmt_reset_upload();
+#elif defined CONFIG_DESKTOP_DFU_BACKEND_SUIT
+	suit_dfu_cleanup();
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu_mcumgr.c:51]: > Check returned value, please.

module state error

File: applications/nrf_desktop/src/util/desktop_suit.c (hunk 18 above)
Comment (mkapala-nordic) [reply applications/nrf_desktop/src/util/desktop_suit.c]: Ignore as in teams discussion it was decided that this util will be removed.

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf [hunk 23]
```diff
@@ -107,3 +107,31 @@ CONFIG_LOG_MODE_DEFERRED=y
 CONFIG_LOG_PRINTK=y
 CONFIG_LOG_BUFFER_SIZE=4096
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
+
+# Enable MCUmgr Bluetooth transport
+CONFIG_DESKTOP_DFU_MCUMGR_ENABLE=y
```
[the remark this answers is not in the document]
Comment (zycz) [reply applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:112]: Adding only to debug conf as we don't have logs in release conf.

**Review by mkapala-nordic — APPROVED** [review]

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf (hunk 23 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:114]: Sure

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr [hunk 24]
```diff
@@ -9,20 +9,14 @@ menuconfig DESKTOP_DFU_MCUMGR_ENABLE
...
-	select IMG_MANAGER
-	select MCUBOOT_BOOTUTIL_LIB
-	select STREAM_FLASH
-	select FLASH_MAP
+	select NET_BUF
 	select FLASH
-	depends on BOOTLOADER_MCUBOOT
+	select FLASH_MAP
+	select ZCBOR
+
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:19]: Remove extra empty line, please

File: applications/nrf_desktop/src/modules/dfu_mcumgr.c [hunk 25]
```diff
@@ -110,6 +126,41 @@ static struct mgmt_callback cmd_recv_cb = {
...
+	int err = suit_get_supported_manifest_info(SUIT_MANIFEST_APP_ROOT, &class_info);
+
+	if (!err) {
+		err = suit_get_installed_manifest_info(&(class_info.class_id),
+				&seq_num, NULL, NULL, NULL, NULL);
+	}
+	if (!err) {
+		LOG_INF("SUIT sequence number: %d", seq_num);
+	} else {
+		LOG_ERR("suit_get_installed_manifest_seq_num failed (err: %d)", err);
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/dfu_mcumgr.c:157]: The error log seems outdated

**Review by MarekPieta — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Review by mkapala-nordic — APPROVED** [review]

**Review by tomchy — APPROVED** [review]
